interchain-token-service/src/state.rs

```diff
@@ -0,0 +1,127 @@
+use std::collections::HashMap;
+
+use axelar_wasm_std::IntoContractError;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, StdError, Storage};
+use cw_storage_plus::{Item, Map};
+use router_api::{Address, ChainName};
+
+#[derive(thiserror::Error, Debug, IntoContractError)]
+pub enum Error {
+    #[error(transparent)]
+    Std(#[from] StdError),
+    #[error("ITS contract got into an invalid state, its config is missing")]
+    MissingConfig,
+    #[error("its address for chain {0} not found")]
+    ItsAddressNotFound(ChainName),
+}
+
+#[cw_serde]
+pub struct Config {
+    pub axelarnet_gateway: Addr,
+}
+
+const CONFIG: Item<Config> = Item::new("config");
+const ITS_ADDRESSES: Map<&ChainName, Address> = Map::new("its_addresses");
+
+pub fn load_config(storage: &dyn Storage) -> Result<Config, Error> {
+    CONFIG.may_load(storage)?.ok_or(Error::MissingConfig)
+}
```

Comment:
something to discuss: in the axelarnet-gateway I ended up making this panic if the config isn't there and just return `Config`. Missing the config means that either instantiation or migration went wrong, i.e. there is a bug in the code.

Reply:
sounds good to me